src/main/drivers/serial_uart_stm32f30x.c

```diff
@@ -143,11 +215,16 @@ void serialUARTInit(IO_t tx, IO_t rx, portMode_t mode, portOptions_t options, ui
 #ifdef USE_UART1
 uartPort_t *serialUART1(uint32_t baudRate, portMode_t mode, portOptions_t options)
 {
-    uartPort_t *s;
     static volatile uint8_t rx1Buffer[UART1_RX_BUFFER_SIZE];
     static volatile uint8_t tx1Buffer[UART1_TX_BUFFER_SIZE];
 
-    s = &uartPort1;
+    uartDevice_t *uart = uartHardwareMap[0];
+    if (!uart)
+        return NULL;
+
+    uartPort_t *s = &uartPort1;     // original (OK)
```

Comment:
@ledvinap 
I'm having a hard time finding out what I'm doing wrong with this attempt.

I have a working `drivers/serial_uart_stm32f4xx.c`, and starting to convert `drivers/serial_uart_stmf30x.c`.
As you can see, I'm just about started to modify `serialUART1`, and as soon as I try to use `&(uart->port)` instead of `&uartPort1`, the FC wedges when UART1 is opened (e.g. MSP).

Would you care to take some time to take a look at this with fresh eyes?